src/common/motor.rs

```diff
@@ -161,8 +164,51 @@ impl TryFrom<Kind> for MotorPinsConfig {
                 }
             },
         };
-
-        Ok(Self { a, b, dir, pwm })
+        let enable_pin_high = match value.get("en_high") {
+            Ok(opt) => match opt {
+                Some(val) => Some(val.try_into()?),
+                None => None,
+            },
+            Err(err) => match err {
+                AttributeError::KeyNotFound(_) => None,
+                _ => {
+                    return Err(err);
+                }
+            },
+        };
+        let enable_pin_low = match value.get("en_low") {
+            Ok(opt) => match opt {
+                Some(val) => Some(val.try_into()?),
+                None => None,
+            },
+            Err(err) => match err {
+                AttributeError::KeyNotFound(_) => None,
+                _ => {
+                    return Err(err);
+                }
+            },
+        };
+        let pwm_frequency = match value.get("pwm_freq") {
```

Comment:
can write it in the more functional way.
```
value
            .get("pwm_freq")
            .and_then(|v| v.map(TryInto::try_into).transpose())
            .or_else(|e| {
                if matches!(e, AttributeError::KeyNotFound(_)) {
                    Ok(None)
                } else {
                    Err(e)
                }
            });
```
We are going to remove `AttributeError::KeyNotFound` because returning none is functionally the same so the or_else will be removed in the future